src/components/account/AccountSetting.jsx

```diff
@@ -1,46 +1,38 @@
 import { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  setUserData,
+  setUserDataPhoneNumber,
+} from "../../utils/slice/accountsSlice.js";
 import axios from "axios";
 import { useParams } from "react-router-dom";
 import { formatPhoneNumber } from "../../utils/formatPhoneNumber.js";
 
 const AccountSetting = () => {
   const { id } = useParams();
+  const dispatch = useDispatch();
+  const userData = useSelector((state) => state.accounts.userData);
   const [phoneNumber, setPhoneNumber] = useState("");
-  const [userData, setUserData] = useState({
-    id: "",
-    google_account_id: "",
-    contact_preferences: "",
-    email: "",
-    first_name: "",
-    last_name: "",
-    role: "",
-    phone_number: "",
-  });
 
   useEffect(() => {
     const fetchUserData = async () => {
       try {
         const response = await axios.get(`http://localhost:3000/user/${id}`);
```

Comment:
FYI, I'm storing my axios requests in the utils folder. Up to you if you want to move them over